### ISSUE TITLE
Rule Change: Add check for missing spaces before closing hashes in ATX headings

### ISSUE THREAD

**Issue by xbinaryx** [eslint/markdown#409]
### What rule do you want to change?

no-missing-atx-heading-space

### What change do you want to make?

Generate more warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```markdown
#Heading 1#
```

### What does the rule currently do for this code?

Currently, the rule only reports an error for `#Heading` (missing space after opening hash) but ignores `1#` (missing space before closing hash).

### What will the rule do after it's changed?

After the change, the rule will report errors for missing space before closing hash

### Participation

- [x] I am willing to submit a pull request to implement this change.

### Additional comments

This behavior aligns with Markdownlint's implementation (rule MD018), which also checks for spaces both before and after hash sequences in ATX headings.

**Comment by snitin315** [comment]
I'm not sure if we need an option if this is just a stylistic choice, that is not the scope for this plugin

**Comment by lumirlumir** [comment]
Considering the previous conversations and reactions, marking this suggestion as accepted.
